src/io.rs

```diff
@@ -128,6 +126,28 @@ impl Write for UdpWriteAdapter {
     }
 }
 
+/// Adapter for writing to a `UdsSocket` via the `Write` trait
```

Comment:
Since `UnixDatagram` is Unix specific, the `uds` module is going to be behind a `#[cfg]` directive. Can this adapter be moved there to make that easier?

Reply:
Moved. I honestly think it makes more sense there and just put it here for consistency.

Reply:
Yeah, I think you're right. I should probably move the UDP adapter as well.